src/bokeh/models/tools.py

```diff
@@ -701,6 +701,28 @@ class CrosshairTool(InspectTool):
     def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
 
+    overlay = Either(
+        Auto,
+        Instance(Span),
+        Tuple(Instance(Span), Instance(Span)), default="auto", help="""
+    An annotation drawn to indicate the crosshair.
+
+    If ``"auto"``, this will create spans depending on the ``dimensions``
+    property, which based on its value, will result in either one span
+    (horizontal or vertical) or two spans (horizontal and vertical).
+
+    Alternatively the user can provide one ``Span`` instance, where the
```

Comment:
What happens if both are provided, and they disagree

Reply:
Then they will disagree or overlap. It's up to the user to set up a sensible configuration. In fact, in future this may be a feature to have multiple spans or perhaps some other form of annotation on a the same line, to e.g. allow to have an overlay left and right of a data point, but not cover the data point itself. In that case tuple could be generalized into an array, in a fully generic setup, this was be handled by a renderer group.